docs/resources/cdm_cluster_action.md

```diff
@@ -0,0 +1,112 @@
+---
+subcategory: "Cloud Data Migration (CDM)"
+---
+
+# huaweicloud_cdm_cluster_action
+
+Manages a CDM cluster action resource within HuaweiCloud.  
+
+## Example Usage
+
+### Restart service process immediately
+
+```hcl
+variable "cdm_cluster_id" {}
+
+resource "huaweicloud_cdm_cluster_action" "restart" {
+  cluster_id = var.cdm_cluster_id
+  type       = "restart"
+
+  restart {
+    level      = "SERVICE"
+    mode       = "IMMEDIATELY"
+  }
+}
+```
+
+### Restart vm
+
+```hcl
+variable "cdm_cluster_id" {}
+
+resource "huaweicloud_cdm_cluster_action" "restart" {
+  cluster_id = var.cdm_cluster_id
+  type       = "restart"
+
+  restart {
+    level      = "VM"
+    mode       = "FORCIBLY"
+    delay_time = 0
+  }
+}
+```
+
+### Start a cluster
+
+```hcl
+variable "cdm_cluster_id" {}
+
+resource "huaweicloud_cdm_cluster_action" "start" {
+  cluster_id = var.cdm_cluster_id
+  type       = "start"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the resource.
+  If omitted, the provider-level region will be used. Changing this parameter will create a new resource.
+
+* `cluster_id` - (Required, String, ForceNew) ID of CDM cluster.  
+
+  Changing this parameter will create a new resource.
+
+* `type` - (Required, String, ForceNew) Action type.  
```

Comment:
what will happen when starting the cluster which is already started?

Reply:
throw an error